src/Apps/Order/Routes/Payment/index.tsx

```diff
@@ -44,32 +45,59 @@ export const ContinueButton = props => (
 
 export class PaymentRoute extends Component<PaymentProps, PaymentState> {
   state = {
-    address: {
-      ...emptyAddress,
-      country: "US",
-    },
     hideBillingAddress: true,
     error: null,
     isComittingMutation: false,
   }
 
-  onContinueButtonPressed = () => {
-    const billingAddress = this.getSelectedBillingAddress()
+  get startingAddress(): Address {
+    return {
+      ...emptyAddress,
+      country: "US",
+      // shipping form only
+      // ...pick(this.props.order.requestedFulfillment, Object.keys(emptyAddress)),
+    }
+  }
+
+  onContinue = (address: Address) => {
+    const stripeBillingAddress = this.getStripeBillingAddress(address)
 
     this.setState({ isComittingMutation: true }, () => {
-      this.props.stripe.createToken(billingAddress).then(({ error, token }) => {
-        if (error) {
-          this.setState({
-            error,
-            isComittingMutation: false,
-          })
-        } else {
-          this.createCreditCard({ token: token.id })
-        }
-      })
+      this.props.stripe
+        .createToken(stripeBillingAddress)
+        .then(({ error, token }) => {
+          if (error) {
+            this.setState({
+              error,
+              isComittingMutation: false,
+            })
+          } else {
+            this.createCreditCard({ token: token.id })
+          }
+        })
     })
   }
 
+  validateAddress = values => {
+    if (!this.state.hideBillingAddress) {
```

Comment:
this should probably follow the same check as `getSelectedBillingAddress()` below that also factors in fulfillment type.